business_rules/operators.py

```diff
@@ -553,21 +553,41 @@ def not_matches_regex(self, other_value):
     def starts_with(self, other_value):
         target = self.replace_prefix(other_value.get("target"))
         comparator = other_value.get("comparator")
-        results = self.value[target].str.startswith(comparator)
+        value_is_literal: bool = other_value.get("value_is_literal", False)
+        comparison_data: Union[str, pd.Series] = self.get_comparator_data(comparator, value_is_literal)
+        if isinstance(comparison_data, pd.Series):
+            results = self.value.apply(lambda row: row[target].startswith(comparison_data[row.name]), axis=1)
+        else:
+            results = self.value[target].str.startswith(comparison_data)
         return pd.Series(results.values)
 
     @type_operator(FIELD_DATAFRAME)
     def ends_with(self, other_value):
         target = self.replace_prefix(other_value.get("target"))
         comparator = other_value.get("comparator")
-        results = self.value[target].str.endswith(comparator)
+        value_is_literal: bool = other_value.get("value_is_literal", False)
+        comparison_data: Union[str, pd.Series] = self.get_comparator_data(comparator, value_is_literal)
+        if isinstance(comparison_data, pd.Series):
+            results = self.value.apply(lambda row: row[target].endswith(comparison_data[row.name]), axis=1)
```

Comment:
Wouldn't these 'apply' methods be faster as np.where functions:
`np.where(self.value[target].str.endswith(self.value[comparator])`

Reply:
Good idea, will update